components/update_client/protocol_serializer_json.cc

```diff
@@ -19,157 +19,159 @@ using Value = base::Value;
 
 std::string ProtocolSerializerJSON::Serialize(
     const protocol_request::Request& request) const {
-  Value root_node(Value::Type::DICTIONARY);
-  auto* request_node =
-      root_node.SetKey("request", Value(Value::Type::DICTIONARY));
-  request_node->SetKey("protocol", Value(request.protocol_version));
-  request_node->SetKey("dedup", Value("cr"));
-  request_node->SetKey("acceptformat", Value("crx2,crx3"));
+  base::Value::Dict root_node;
+  base::Value::Dict request_node;
+  request_node.Set("protocol", request.protocol_version);
+  request_node.Set("dedup", "cr");
+  request_node.Set("acceptformat", Value("crx2,crx3"));
   if (!request.additional_attributes.empty()) {
     for (const auto& attr : request.additional_attributes)
-      request_node->SetKey(attr.first, Value(attr.second));
+      request_node.Set(attr.first, attr.second);
   }
-  request_node->SetKey("sessionid", Value(request.session_id));
-  request_node->SetKey("requestid", Value(request.request_id));
-  request_node->SetKey("@updater", Value(request.updatername));
-  request_node->SetKey("prodversion", Value(request.updaterversion));
-  request_node->SetKey("updaterversion", Value(request.prodversion));
-  request_node->SetKey("lang", Value(request.lang));
```

Comment:
Some of the changes here are not only about clean up base::Value. I guess it's directly from upstream code. However, we didn't do a fully update for this ``update_client``. Can you confirm this change is correct?

Reply:
Thanks for finding this, I tried to keep the behavior the same, but this key fell through the cracks—seems like upstream it's set for the app_node instead. I'll add STARBOARD macro checks and set it for the request_node instead as is currently done in trunk